components/new_post/post_component.tsx

```diff
@@ -0,0 +1,608 @@
+import PostBodyAdditionalContent from 'components/post_view/post_body_additional_content';
+import PostMessageContainer from 'components/post_view/post_message_view';
+import {getDateForUnixTicks, makeIsEligibleForClick} from 'utils/utils';
+import {getHistory} from 'utils/browser_history';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+import CommentedOn from 'components/post_view/commented_on/commented_on';
+
+import {UserProfile} from '@mattermost/types/users';
+
+import PriorityLabel from 'components/post_priority/post_priority_label';
+
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
+    currentUserId: string;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isFlagged: boolean;
+    previewCollapsed?: string;
+    previewEnabled?: boolean;
+    isEmbedVisible?: boolean;
+    enableEmojiPicker?: boolean;
+    enablePostUsernameOverride?: boolean;
+    isReadOnly?: boolean;
+    pluginPostTypes?: {[postType: string]: PostPluginComponent};
+    channelIsArchived?: boolean;
+    isConsecutivePost?: boolean;
+    isLastPost?: boolean;
+    recentEmojis: Emoji[];
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    channelName?: string;
+    displayName: string;
+    teamDisplayName?: string;
+    teamName?: string;
+    channelType?: string;
+    a11yIndex?: number;
+    isBot: boolean;
+    hasReplies?: boolean;
+    isFirstReply?: boolean;
+    previousPostIsComment?: boolean;
+    matches?: string[];
+    term?: string;
+    isMentionSearch?: boolean;
+    location: keyof typeof Locations;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+        removePost: (post: Post) => void;
+        closeRightHandSide: () => void;
+        selectPostCard: (post: Post) => void;
+        setRhsExpanded: (rhsExpanded: boolean) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isCollapsedThreadsEnabled?: boolean;
+    isMobileView: boolean;
+    canReply?: boolean;
+    replyCount?: number;
+    isFlaggedPosts?: boolean;
+    isPinnedPosts?: boolean;
+    clickToReply?: boolean;
+    isCommentMention?: boolean;
+    parentPost?: Post;
+    parentPostUser?: UserProfile | null;
+    shortcutReactToLastPostEmittedFrom?: string;
+    isPostAcknowledgementsEnabled: boolean;
+    isPostPriorityEnabled: boolean;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const isSearchResultItem = (props.matches && props.matches.length > 0) || props.isMentionSearch || (props.term && props.term.length > 0);
```

Comment:
@koox00 tested this change but for some issues like with showing the thread header, the location will change but its still a search result item, so that is needed for now :/